python/lsst/cp/pipe/ptc.py

```diff
@@ -792,282 +761,6 @@ def covDirectValue(self, diffImage, weightImage, dx, dy):
-            amplifier.
-        """
-        datasetNonLinearity = {ampName: [] for ampName in datasetPtc.ampNames}
-        for i, ampName in enumerate(datasetPtc.ampNames):
-            # If a mask is not found, use all points.
-            if (len(datasetPtc.visitMask[ampName]) == 0):
-                self.log.warn(f"Mask not found for {ampName} in non-linearity fit. Using all points.")
-                mask = np.repeat(True, len(datasetPtc.rawExpTimes[ampName]))
-            else:
-                mask = datasetPtc.visitMask[ampName]
-
-            timeVecFinal = np.array(datasetPtc.rawExpTimes[ampName])[mask]
-            meanVecFinal = np.array(datasetPtc.rawMeans[ampName])[mask]
-
-            # Non-linearity residuals (NL of mean vs time curve): percentage, and fit to a quadratic function
-            # In this case, len(parsIniNonLinearity) = 3 indicates that we want a quadratic fit
-            datasetLinRes = self.calculateLinearityResidualAndLinearizers(timeVecFinal, meanVecFinal)
-
-            # LinearizerLookupTable
-            if tableArray is not None:
-                tableArray[i, :] = datasetLinRes.linearizerTableRow
-
-            datasetNonLinearity[ampName] = datasetLinRes
-
-        return datasetNonLinearity
-
-    def calculateLinearityResidualAndLinearizers(self, exposureTimeVector, meanSignalVector):
-        """Calculate linearity residual and fit an n-order polynomial to the mean vs time curve
-        to produce corrections (deviation from linear part of polynomial) for a particular amplifier
-        to populate LinearizeLookupTable.
-        Use the coefficients of this fit to calculate the correction coefficients for  LinearizePolynomial
-        and LinearizeSquared."
-
-        Parameters
-        ---------
-
-        exposureTimeVector: `list` of `float`
-            List of exposure times for each flat pair
-
-        meanSignalVector: `list` of `float`
-            List of mean signal from diference image of flat pairs
-
-        Returns
-        -------
-        dataset : `lsst.cp.pipe.ptc.LinearityResidualsAndLinearizersDataset`
-            The dataset containing the fit parameters, the NL correction coefficients, and the
-            LUT row for the amplifier at hand.
-
-        Notes
-        -----
-        datase members:
-
-        dataset.polynomialLinearizerCoefficients : `list` of `float`
-            Coefficients for LinearizePolynomial, where corrImage = uncorrImage + sum_i c_i uncorrImage^(2 +
-            i).
-            c_(j-2) = -k_j/(k_1^j) with units DN^(1-j) (c.f., Eq. 37 of 2003.05978). The units of k_j are
-            DN/t^j, and they are fit from meanSignalVector = k0 + k1*exposureTimeVector +
-            k2*exposureTimeVector^2 + ... + kn*exposureTimeVector^n, with
-            n = "polynomialFitDegreeNonLinearity". k_0 and k_1 and degenerate with bias level and gain,
-            and are not used by the non-linearity correction. Therefore, j = 2...n in the above expression
```

Comment:
It seems that this explanation and reference got lost, but I think it helped the user understand a bit better how the polynomial correction was constructed. Could this (or a summary) be placed back somewhere? Or maybe we could still document this in the comments of the Jira ticket, at least.

Reply:
Added and updated this discussion in the Notes section of the run method docstring.